test/e2e/main_test.go

```diff
@@ -2,10 +2,80 @@ package e2e
 
 import (
 	"testing"
+	"time"
 
+	"github.com/openshift/windows-machine-config-bootstrapper/tools/windows-node-installer/pkg/types"
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachineconfig/tracker"
 	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
+var (
+	// nrNodes represent the number of nodes to be dealt with in the test suite.
+	// Expose this as a flag.
+	nrNodes = 1
+	// gc holds the global context across the test suites.
+	gc = globalContext{numberOfNodes: 1, windowsVMs: make([]types.WindowsVM, 0, nrNodes)}
+)
+
+// globalContext holds the information that we want to use across the test suites.
+// If you want to move item here make sure that
+// 1.) It is needed across test suites
+// 2.) You're responsible for checking if the field is stale or not. Any field
+//     in this struct is not guaranteed to be latest from the apiserver.
+type globalContext struct {
+	// windowsVMs has the slices of WindowsVMs that are created by WMCO. These are
+	// populated from the secrets generated by WMCO.
+	windowsVMs []types.WindowsVM
+	// numberOfNodes to be used for the test suite.
+	numberOfNodes int
+}
+
+// testContext holds the information related to the individual test suite. This data structure
+// should be instantiated by every test suite, so that we can update the test context to be
+// passed around to get the information which was created within the test suite. For example,
+// if the create test suite creates a Windows Node object, the node object and other related
+// information should be easily accessible by other methods within the same test suite.
+// Some of the fields we have here can be exposed by via flags to the test suite.
+type testContext struct {
+	// namespace holds the namespace of
+	namespace string
+	// osdkTestCtx holds the information related to operator sdk framework's test Context
+	osdkTestCtx *framework.TestCtx
+	// nodes holds the list of nodes
+	nodes []v1.Node
+	// credentials to be used to access the node.
+	credentials []tracker.Credentials
+	// kubeclient holds the kube client
+	kubeclient kubernetes.Interface
+	// tracketConfigmap holds the information related to the configmap created by controller
```

Comment:
It does not hold this information but is the (or pointer to if you want to be pedantic) `ConfigMap` created by the operator. Please be consistent and use operator.